events/wc_offmap_events.txt

```diff
@@ -642,41 +664,180 @@ character_event = {
+						limit = {
+							has_wonder_upgrade = upgrade_old_god_cthun_1
+						}
+						destroy_wonder_upgrade = upgrade_old_god_cthun_1
+						offmap_cthun = {
+							set_status = old_god_dead
+						}
+						ROOT = { set_character_flag = killed_cthun }
+					}
+					if = {
+						limit = {
+							has_wonder_upgrade = upgrade_old_god_cthun_2
+						}
+						destroy_wonder_upgrade = upgrade_old_god_cthun_2
+						offmap_cthun = {
+							set_status = old_god_dead
+						}
+						ROOT = { set_character_flag = killed_cthun }
+					}
+					if = {
+						limit = {
+							has_wonder_upgrade = upgrade_old_god_nzoth_1
+						}
+						destroy_wonder_upgrade = upgrade_old_god_nzoth_1
+						offmap_nzoth = {
+							set_status = old_god_dead
+						}
+						ROOT = { set_character_flag = killed_nzoth }
+					}
+					if = {
+						limit = {
+							has_wonder_upgrade = upgrade_old_god_nzoth_2
+						}
+						destroy_wonder_upgrade = upgrade_old_god_nzoth_2
+						offmap_nzoth = {
+							set_status = old_god_dead
+						}
+						ROOT = { set_character_flag = killed_nzoth }
+					}
+					if = {
+						limit = {
+							has_wonder_upgrade = upgrade_old_god_yogg_saron_1
+						}
+						destroy_wonder_upgrade = upgrade_old_god_yogg_saron_1
+						offmap_yoggsaron = {
+							set_status = old_god_dead
+						}
+						ROOT = { set_character_flag = killed_yoggsaron }
+					}
+					if = {
+						limit = {
+							has_wonder_upgrade = upgrade_old_god_yogg_saron_2
+						}
+						destroy_wonder_upgrade = upgrade_old_god_yogg_saron_2
+						offmap_yoggsaron = {
+							set_status = old_god_dead
+						}
+						ROOT = { set_character_flag = killed_yoggsaron }
+					}
+				}
+			}
 		}
 	}
 
 	option = {
 		name = EVTOPTA_WCOME.200
 		prestige = 1000
-		give_nickname = nick_the_godslayer
-        if = {
+		if = {
+			limit = { NOT = { has_nickname = nick_the_godslayer } }
+			give_nickname = nick_the_godslayer
+		}
+		if = {
+			limit = {
+				has_character_flag = killed_cthun
+				has_character_flag = killed_nzoth
+				has_character_flag = killed_yoggsaron
+			}
+			give_nickname = nick_twilight_of_the_gods
+		}
+		if = {
             limit = {
-                is_female = yes
+                has_character_flag = killed_cthun
+				NOT = { has_character_flag = killed_cthun_block }
             }
-            create_bloodline = {
-                type = godslayer_cthun
-                inheritance = matrilineal
+			if = {
+				limit = {
+					is_female = yes
+				}
+				create_bloodline = {
+					type = godslayer_bloodliner
+					inheritance = matrilineal
+				}
+			}
+			else = {
+				create_bloodline = {
+					type = godslayer_bloodline
+				}
+			}
+			new_bloodline = {
+				set_description = godslayer_cthun_desc
+			}
+			set_character_flag = killed_cthun_block
+			set_character_flag = killed_cthun_desc
+        }
+		else_if = {
+            limit = {
+                has_character_flag = killed_nzoth
+				NOT = { has_character_flag = killed_nzoth_block }
```

Comment:
So, what's a point of the flag? You could just tie it to the bloodlines.

Reply:
it checks the character that founded the bloodline and it seemed to me that its the easiest way to do it

Reply:
There's `is_bloodline_founder_of` condition.
https://ck2.paradoxwikis.com/Conditions

Reply:
how do i check which bloodline is it cause they all use the same type?